docs/quickstart.rst

```diff
@@ -61,198 +199,276 @@ instance had already existing in the destination directory, this command
+
+For ``pg_hba.conf`` on the monitor node pg_autoctl inspects the local network
+and makes its best guess about the subnet to allow. In our case it guessed
+correctly:
+
+.. code-block:: ini
+
+   # automatically added to the monitor
+
+   host "pg_auto_failover" "autoctl_node" 10.0.1.0/24 trust
+
+If worker nodes have more ad-hoc addresses and are not in the same subnet, it's
+better to disable pg_autoctl's automatic modification of pg_hba using the
+``--skip-pg-hba`` command line option during creation. You will then need to
+edit the hba file by hand. Another reason for manual edits would be to use
+special authentication methods.
 
 Watch the replication
 ---------------------
 
 First let’s verify that the monitor knows about our nodes, and see what
 states it has assigned them:
 
-.. code-block:: text
+.. code-block::
 
-   pg_autoctl show state --pgdata ./monitor
-        Name |   Port | Group |  Node |     Current State |    Assigned State
-   ----------+--------+-------+-------+-------------------+------------------
-   127.0.0.1 |   6010 |     0 |     1 |           primary |           primary
-   127.0.0.1 |   6011 |     0 |     2 |         secondary |         secondary
+   ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
 
-This looks good. We can add data to the primary, and watch it get
-reflected in the secondary.
+                              Name |   Port | Group |  Node |     Current State |    Assigned State
+   --------------------------------+--------+-------+-------+-------------------+------------------
+   ha-demo-a.internal.cloudapp.net |   5432 |     0 |     1 |           primary |           primary
+   ha-demo-b.internal.cloudapp.net |   5432 |     0 |     2 |         secondary |         secondary
+
+
+This looks good. We can add data to the primary, and later see it appear in the
+secondary. We'll connect to the database from inside our "app" virtual machine,
+using a connection string obtained from the monitor. First we'll get the
+connection string and store it in a local environment variable:
 
 .. code-block:: bash
 
-   # add data to primary
-   psql -p 6010 \
-     -c 'create table foo as select generate_series(1,1000000) bar;'
+   APP_DB_URI=$( \
+     ssh -l ha-admin `vm_ip monitor` \
+       pg_autoctl show uri --formation default --pgdata monitor \
+   )
 
-   # query secondary
-   psql -p 6011 -c 'select count(*) from foo;'
-     count
-   ---------
-    1000000
+The connection string contains both our nodes, comma separated, and includes
+the url parameter ``?target_session_attrs=read-write`` telling psql that we
+want to connect to whichever of these servers supports reads *and* writes.
+That will be the primary server.
+
+.. code-block:: bash
+
+   # connect to database via psql on the app vm and
+   # create a table with a million rows
+   ssh -l ha-admin -t `vm_ip app` -- \
+     psql "\"$APP_DB_URI\"" \
+       -c "\"CREATE TABLE foo AS SELECT generate_series(1,1000000) bar;\""
```

Comment:
Does not work for me:

```
$ ssh -l ha-admin -t `vm_ip app` --   psql "'$APP_DB_URI'" -c '"CREATE TABLE foo AS SELECT generate_series(1,1000000) bar;"'
psql: invalid sslmode value: ""
Connection to 20.188.42.21 closed.

$ ssh -l ha-admin `vm_ip monitor` -- pg_autoctl show uri --pgdata monitor --formation default
postgres://ha-demo-a.internal.cloudapp.net:5432,ha-demo-b.internal.cloudapp.net:5432/appdb?target_session_attrs=read-write&sslmode=

$ export APP_DB_URI='postgres://ha-demo-a.internal.cloudapp.net:5432,ha-demo-b.internal.cloudapp.net:5432/appdb?target_session_attrs=read-write&sslmode=require' 

$ ssh -l ha-admin -t `vm_ip app` -- \
>   psql "\"$APP_DB_URI\"" \
>     -c "\"CREATE TABLE foo AS SELECT generate_series(1,1000000) bar;\""
psql: FATAL:  no pg_hba.conf entry for host "10.0.1.5", user "ha-admin", database "appdb", SSL on
Connection to 20.188.42.21 closed.
```

  1. This looks like a bug in pg_autoctl itself, we should fix it and/or make sure it's fixed in 1.3
  2. the HBA has not been edited for the application VM and its IP address, and that's not done by `pg_autoctl` itself. I think now is a good time in this documentation to include how to achieve that: users will need to.

Reply:
Another way to quote it:

```
ssh -l ha-admin -t `vm_ip a` --   psql "'$APP_DB_URI'"     -c "'CREATE TABLE foo AS SELECT generate_series(1,1000000) bar;'"
```

Reply:
> This looks like a bug in pg_autoctl itself, we should fix it and/or make sure it's fixed in 1.3

I tried with 1.3 today and ran into the same problem. I was able to run the command through node `vm_ip a` but not `vm_ip app`. The latter says:

```
psql: FATAL:  no pg_hba.conf entry for host "10.0.1.4", user "ha-admin", database "appdb", SSL on
```

How should we proceed? Add instructions to modify pg_hba by hand, make a new release of failover to fix the problem, or run the command ssh'd through node A?

Reply:
Yes, the HBA needs to be manually edited by our users to add in their application nodes. At the moment this needs to be done on both the primary and the secondary. It is possible to do it when there's only a primary, and then `pg_basebackup` will cary the changes to the secondary when creating it.